sdk/identity/azure-identity/azure/identity/_credentials/auth_file.py

```diff
@@ -0,0 +1,48 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import six
+import json
+
+from .client_credential import ClientSecretCredential
+from azure.core.exceptions import ClientAuthenticationError
+
+
+def _parse_credentials_file(file_path):
+    with open(file_path, 'rb') as f:
+        auth_str = six.ensure_str(f.read())
+        
+    return json.loads(auth_str)
+
+
+class AuthFileCredential(object):
+
+    def __init__(self, file_path, **kwargs):
+        self._credential = None
+        self.file_path = file_path
+        self.kwargs = kwargs
+
+    def get_token(self, *scopes, **kwargs):
+        self._ensure_credential()
+        return self._credential.get_token(*scopes, **kwargs)
+
+    def _ensure_credential(self):
+        if self._credential is None:
+            try:
+                self.kwargs.pop('activeDirectoryEndpointUrl', None)
```

Comment:
?

Reply:
removed this line

Reply:
Do we pop `self._kwargs['authority']` (wrong key name `activeDirectoryEndpointUrl` here in the first commit, sorry) to prevent repeating argument twice when creating the inner credential?
Or raise error if `authority` supplied when instantiating an `AuthFileCredential`?
Because we pass in `self._kwargs` when creating the inner credential, but `authority` already specified, and the its value is from the auth file.
Currently I simply just removed the `pop` line.